File: python/02-linearSearch/search-_in_range.py
# search in range

def search_in_range(arr, target, start, end):
    if len(arr) == 0:
        return -1
    
    for i in range(start, end + 1):
        if arr[i] == target:
            return i
    return -1

arr = [1, 2, 3, 4, 5, 6, 7, 8, 9]
print(search_in_range(arr, 8, 0, 5))

File: javascript/01-arrays/searchInRange.js
function linearSearch1(arr, target, start, end) {
    if (arr.length === 0) {
        return -1;
    }
    for (let i = start; i <= end; i++) {
        if (arr[i] === target) {
            return i;
        }
    }
    return -1;
}
const arr = [18, 12, -7, 3, 14, 28];
const target = 14;
console.log(linearSearch1(arr, target, 1, 4));
